src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IepLevelService.kt

```diff
@@ -24,7 +24,7 @@ class IepLevelService(
 
       withContext(Dispatchers.Default) {
         iepPrisonRepository.findAllByPrisonIdAndActiveIsTrue(prisonId)
-          .filter { iepLevelMap[it.iepCode] != null && iepLevelMap[it.iepCode]!!.active }
```

Comment:
@neilmendum it's interesting despite the `iepLevelMap[it.iepCode] != null &&` the Kotlin compiler still doesn't seem to infer that `iepLevelMap[it.iepCode]` is now a `Boolean` (not a `Boolean?`).

We should definitely [try and avoid using `!!`](https://kotlinlang.org/docs/null-safety.html#the-operator) when an alternative is available as we can avoid these `NullPointerException` this way (I guess I'm not one of those "NPE lovers" :P)

Reply:
PS: I'll update the test to add a case for when a prison have some "weird" level which is not present in the global `IepLevel` table to cover the `iepLevelMap[it.iepCode] == null` case.

Reply:
New test: https://github.com/ministryofjustice/hmpps-incentives-api/pull/149/files#diff-fc67634b9869e5d81f9583cbfd10674aaa100a11d0e83b97fcae6dc57e844dc1